ethcore/src/client/client.rs

```diff
@@ -2191,6 +2192,13 @@ impl IoClient for Client {
 				let first = queued.write().1.pop_front();
 				if let Some((unverified, receipts_bytes)) = first {
 					let hash = unverified.hash();
+					let parent_hash = unverified.parent_hash();
+					let status = client.importer.block_queue.status(&parent_hash);
```

Comment:
Can you add a comment for why we are checking this? I understood after you explained the scenario, but it was not obvious to me reading the code initially.

Reply:
On second thoughts would it not be better to replace 

https://github.com/paritytech/parity-ethereum/blob/18fdcb429e514dc64f1ce0b16399768432a8b235/ethcore/src/client/client.rs#L2170-L2171

with `!self.chain.read().is_known(&parent_hash)`?  Currently `self.block_status` would indeed return `Known` if the block is in the normal queue, and then it would be rejected later by your check.

Reply:
Yeah sounds like a better idea. In this scenario it would probably better to fail and let syncing to redo ancient block importing from a known good point completely.